js/foundation.dropdown.js

```diff
@@ -178,7 +178,7 @@ class Dropdown extends Positionable {
 
       Keyboard.handleKey(e, 'Dropdown', {
         open: function() {
-          if ($target.is(_this.$anchors)) {
+          if ($target.is(_this.$anchors) && !$target.is('input')) {
```

Comment:
Not only `input` should be able to get a `space` without losing the focus. I think we have two solutions:
* determine all the native elements that should be able to receive `space` and make a big filter
* add an option to the `dropdown` element to prevent the keyboard catching